### ISSUE TITLE
[VersionControl] No appears changes of deleted items (Git)

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7107]
Remove some files, nothing appears in the Review and Commit view. 

```
ERROR [2019-02-08 13:14:38Z]: An unhandled exception has occurred. Terminating MonoDevelop? False
System.NullReferenceException: Object reference not set to an instance of an object
at MonoDevelop.UnitTesting.SolutionFolderTestGroup.Dispose () [0x00029] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Services/SolutionFolderTestGroup.cs:68
at MonoDevelop.UnitTesting.UnitTestGroup.UpdateTests () [0x00025] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Services/UnitTestGroup.cs:162
at MonoDevelop.UnitTesting.SolutionFolderTestGroup.OnEntryChanged (System.Object sender, MonoDevelop.Projects.SolutionItemEventArgs e) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/MonoDevelop.UnitTesting/Services/SolutionFolderTestGroup.cs:76
at (wrapper delegate-invoke) .invoke_void_object_SolutionItemChangeEventArgs(object,MonoDevelop.Projects.SolutionItemChangeEventArgs)
at MonoDevelop.Projects.Solution.OnSolutionItemRemoved (MonoDevelop.Projects.SolutionItemChangeEventArgs args) [0x00088] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/Solution.cs:1257
at MonoDevelop.Projects.SolutionFolder.NotifyItemRemovedFromFolder (System.Object sender, MonoDevelop.Projects.SolutionItemChangeEventArgs e, System.Boolean removedFromSolution) [0x00042] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/SolutionFolder.cs:808
at MonoDevelop.Projects.SolutionFolder.OnItemRemoved (MonoDevelop.Projects.SolutionItemChangeEventArgs e, System.Boolean removedFromSolution) [0x00009] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/SolutionFolder.cs:841
at MonoDevelop.Projects.SolutionFolder.NotifyItemRemoved (MonoDevelop.Projects.SolutionFolderItem item, System.Boolean removedFromSolution) [0x00015] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/SolutionFolder.cs:363
at MonoDevelop.Projects.SolutionFolderItemCollection.OnItemsRemoved (System.Collections.Generic.IEnumerable`1[T] items) [0x00029] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/SolutionFolderItemCollection.cs:85`
`at MonoDevelop.Projects.ItemCollection`1[T].RemoveAt (System.Int32 index) [0x00027] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ItemCollection.cs:145
at MonoDevelop.Projects.ItemCollection`1[T].Remove (T item) [0x00021] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/ItemCollection.cs:134`
`at MonoDevelop.Ide.Gui.Pads.ProjectPad.SolutionFolderNodeCommandHandler.DeleteItem () [0x0009f] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/SolutionFolderNodeBuilder.cs:255`
`at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1023`
`at MonoDevelop.Ide.DispatchService+GtkSynchronizationContext+TimeoutProxy.HandlerInternal (System.IntPtr data) [0x00016] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/DispatchService.cs:77`
`ERROR [2019-02-08 13:15:03Z]: Failed to query git status`
`LibGit2Sharp.AmbiguousSpecificationException: More than one file matches the pathspec '.vs/TipCalc'. You can either force a literal path evaluation (GIT_STATUS_OPT_DISABLE_PATHSPEC_MATCH), or use git_status_foreach().`
`at LibGit2Sharp.Core.Proxy.git_status_file (LibGit2Sharp.Core.Handles.RepositoryHandle repo, LibGit2Sharp.Core.FilePath path) [0x00028] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Proxy.cs:2923`
`at LibGit2Sharp.Repository.RetrieveStatus (System.String filePath) [0x00015] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Repository.cs:1610`
`at MonoDevelop.VersionControl.Git.GitRepository.GetFilesVersionInfoCore (LibGit2Sharp.Repository repo, MonoDevelop.VersionControl.Git.GitRevision rev, System.Collections.Generic.List`1[T] localPaths, System.Collections.Generic.List`1[T] versions) [0x00019] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:482`
`at MonoDevelop.VersionControl.Git.GitRepository.GetDirectoryVersionInfo (MonoDevelop.Core.FilePath localDirectory, System.Collections.Generic.IEnumerable`1[T] localFileNames, System.Boolean getRemoteStatus, System.Boolean recursive) [0x001cc] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:443
at MonoDevelop.VersionControl.Git.GitRepository.OnGetVersionInfo (System.Collections.Generic.IEnumerable`1[T] paths, System.Boolean getRemoteStatus) [0x00002] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:383
```

The problems is that something fails notifying the deletion. 
Can reproduce this one sometimes, not always.

> VS bug [#789304](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789304)

**Comment by jsuarezruiz** [comment]
Sent PR: https://github.com/mono/monodevelop/pull/7141

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1481 (master)

Pull Request #7141 merged by: Lluis Sanchez
Author: mono
Commit: 7af7a11d9aacc10e054ff852dcaca505745fcf51 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2176 (release-8.0)

Pull Request #7141 merged by: Lluis Sanchez
Author: mono
Commit: 7af7a11d9aacc10e054ff852dcaca505745fcf51 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2176 (release-8.0)

Pull Request #7141 merged by: Lluis Sanchez
Author: mono
Commit: 7af7a11d9aacc10e054ff852dcaca505745fcf51 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7141 merged by: Lluis Sanchez
Author: mono
Commit: 7af7a11d9aacc10e054ff852dcaca505745fcf51 (xamarin/monodevelop)
